frontend/src/v5/ui/routes/dashboard/projects/federations/editFederationModal/editFederation/editFederation.component.tsx

```diff
@@ -18,58 +18,48 @@
+import IncludeIcon from '@assets/icons/outlined/plus_minimal-outline.svg';
+import { useCallback, useContext } from 'react';
 import { FormattedMessage } from 'react-intl';
+import { IContainer } from '@/v5/store/containers/containers.types';
 import { useContainersData } from '../../../containers/containers.hooks';
-import { IconContainer, IncludeIcon, RemoveIcon } from './editFederation.styles';
+import { SuccessIconContainer, ErrorIconContainer, SuccessButton, ErrorButton } from './editFederation.styles';
 import { ActionButtonProps, EditFederationContainers } from './editFederationContainersList/editFederationContainersList.component';
 import { IconButtonProps } from './editFederationContainersList/editFederationContainersListItem/editFederationContainersListItem.component';
+import { EditFederationContext } from '../editFederationContext';
 
 type EditFederationProps = {
 	federation: IFederation;
-	onContainersChange: (includedContainers) => void;
 };
 
-export const EditFederation = ({ federation, onContainersChange }: EditFederationProps): JSX.Element => {
+export const EditFederation = ({ federation }: EditFederationProps): JSX.Element => {
 	const { containers } = useContainersData();
-	const getContainerById = (id: string) => containers.find((container: IContainer) => container._id === id);
-	const [includedContainers, setIncludedContainers] = useState<IContainer[]>([]);
-	const [availableContainers, setAvailableContainers] = useState<IContainer[]>([]);
+	const { includedContainers, setIncludedContainers } = useContext(EditFederationContext);
 	const isNewFederation = !federation._id;
 	const isCollaboratorFromId = FederationsHooksSelectors.selectHasCollaboratorAccess(federation._id);
-	const isCollaborator = isNewFederation ? true : isCollaboratorFromId;
-
-	useEffect(() => {
-		if (containers.length) {
-			setIncludedContainers(federation.containers.map(getContainerById));
-		}
-	}, [containers]);
-
-	useEffect(() => {
-		onContainersChange(includedContainers.filter(Boolean)); // filter out 'undefined' which are unassigned containers
-		setAvailableContainers(containers.filter((container) => !includedContainers.includes(container)));
-	}, [includedContainers]);
+	const isCollaborator = isNewFederation || isCollaboratorFromId;
+	const availableContainers = containers.filter(({ _id }) => !includedContainers.find((c) => c._id === _id));
 
 	const includeContainer = (container: IContainer) => {
-		setIncludedContainers((existingState) => [...existingState, container]);
+		setIncludedContainers((oldIncludedContainers) => [...oldIncludedContainers, container]);
 	};
 
-	const removeContainer = (containerToRemove: IContainer) => {
-		setIncludedContainers((existingState) => existingState.filter((container) => container !== containerToRemove));
+	const removeContainer = ({ _id }: IContainer) => {
+		setIncludedContainers((oldIncludedContainers) => oldIncludedContainers.filter((container) => container._id !== _id));
 	};
```

Comment:
I think we could get rid of these two and just call  `includeContainers([container])` and  `removeContainers([container])`.